### PR TITLE
 Make all stores singletons

### PR DESCRIPTION
**Pull request by notVitaliy**

### CODE REVIEW

File: src/stores/marker.store.spec.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ xdescribe('MarkerStore', () => {
   let markerStore: MarkerStore
 
   beforeEach(() => {
-    markerStore = new MarkerStore()
+    markerStore = MarkerStore.instance
```
Comment (Shawn8901) [inline src/stores/marker.store.spec.ts]: As its now an singleton this should be not needed anylonger. Just asign it once in beforeAll

File: src/stores/marker.store.ts [hunk 2]
```diff
@@ -1,35 +1,50 @@
...
+  public static get instance(): MarkerStore {
+    if (!this[singleton]) this[singleton] = new MarkerStore(singletonEnforcer)
+    return this[singleton]
+  }
+
   private markers: Map<string, Marker> = new Map()
 
-  public async getNextBackMarker(exchange: string, symbol: string) {
-    const marker = this.getMarker(exchange, symbol)
+  constructor(enforcer: Symbol) {
```
Comment (Shawn8901) [inline src/stores/marker.store.ts]: Mark this constructor as private, its will also enforce on compiler level, that its only accessible via instance call.

File: src/stores/order.store.spec.ts [hunk 3]
```diff
@@ -23,60 +23,61 @@ const order: OrderWithTrades = {
 }
 
 describe('OrderStore', () => {
+  const storeOpts = { exchange: 'test', symbol: 'test', strategy: 'test' }
   let orderStore: OrderStore
 
   beforeEach(() => {
-    orderStore = new OrderStore()
-    orderStore.addSymbol('test', 'test', 'test')
+    orderStore = OrderStore.instance
```
Comment (Shawn8901) [inline src/stores/order.store.spec.ts]: Same like for the other unit test, its a singleton and already constructed, so it should be enough to do this in a beforeAll. State isnt reseted so there should be no difference if i am right.

Can you check this for the other unit tests also? then i have not to copy the same comment multiple time.

File: src/stores/order.store.ts [hunk 4]
```diff
@@ -16,18 +20,29 @@ export enum ORDER_STATE {
...
+    if (!this[singleton]) this[singleton] = new OrderStore(singletonEnforcer)
+    return this[singleton]
+  }
+
   private openOrders: Map<string, Map<string, OrderWithTrades>> = new Map()
   private orderStates: Map<string, Map<string, ORDER_STATE>> = new Map()
-  private readonly sessionId = sessionStore.sessionId
+  private readonly sessionId = SessionStore.instance.sessionId
+
+  constructor(enforcer: Symbol) {
```
Comment (Shawn8901) [inline src/stores/order.store.ts]: Mark constructor as private.
Can you also do this for every other store? then i have not to copy the comment. :-)

**Review by Shawn8901 — CHANGES_REQUESTED** [review]
I like this refactoring.
Main thing to change is to mark the contructor as private. this will eliminate the instance creation on compiler level (for other classes). You could keep the other mechanism if you want, but afterwards it should not be needed.
The unit test thing is just optional hint.

File: src/engine/exchange.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { ExchangeConf, StrategyConf, Base } from '@m8bTypes'
```
Comment (Shawn8901) [inline src/engine/exchange.ts:1]: This file has a compilation error for me.
There is a `{ exchange: this.exchangeName, symbol } ` missing in line 48 and 66.

As hint: if you try a yarn rebuild before commiting you will find those errors.

**Review by Shawn8901 — CHANGES_REQUESTED** [review]

File: src/stores/marker.store.spec.ts (hunk 1 above)
Comment (notVitaliy) [reply src/stores/marker.store.spec.ts]: Good point.

File: src/engine/exchange.ts (hunk 5 above)
Comment (notVitaliy) [reply src/engine/exchange.ts:1]: Good catch. Was relying too much on the tests. 👍

**Review by Shawn8901 — APPROVED** [review]

**Pull request merged**